cvs diff: Diffing transex3
cvs diff: Diffing transex3/inc
cvs diff: Diffing transex3/inc/pch
cvs diff: Diffing transex3/inc/transex3
cvs diff: Diffing transex3/java
cvs diff: Diffing transex3/java/l10nconv
cvs diff: Diffing transex3/java/l10nconv/documentation
cvs diff: Diffing transex3/java/l10nconv/java
cvs diff: Diffing transex3/java/l10nconv/java/com
cvs diff: Diffing transex3/java/l10nconv/java/com/sun
cvs diff: Diffing transex3/java/l10nconv/java/com/sun/star
cvs diff: Diffing transex3/java/l10nconv/java/com/sun/star/tooling
cvs diff: Diffing transex3/java/l10nconv/java/com/sun/star/tooling/DirtyTags
cvs diff: Diffing transex3/java/l10nconv/java/com/sun/star/tooling/converter
cvs diff: Diffing transex3/java/l10nconv/java/com/sun/star/tooling/converter/dtd
cvs diff: Diffing transex3/java/l10nconv/java/com/sun/star/tooling/languageResolver
cvs diff: Diffing transex3/java/receditor
cvs diff: Diffing transex3/java/receditor/java
cvs diff: Diffing transex3/java/receditor/java/transex3
cvs diff: Diffing transex3/java/receditor/java/transex3/controller
cvs diff: Diffing transex3/java/receditor/java/transex3/model
cvs diff: Diffing transex3/java/receditor/java/transex3/view
cvs diff: Diffing transex3/layout
Index: transex3/layout/layoutparse.cxx
===================================================================
RCS file: /cvs/l10n/transex3/layout/layoutparse.cxx,v
retrieving revision 1.2
retrieving revision 1.2.2.3
diff -p -u -u -p -b -w -B -r1.2 -r1.2.2.3
--- transex3/layout/layoutparse.cxx	6 Mar 2008 12:35:12 -0000	1.2
+++ transex3/layout/layoutparse.cxx	4 Apr 2008 09:11:45 -0000	1.2.2.3
@@ -1,3 +1,38 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
 #include "layoutparse.hxx"
 
 #define STRING( str ) String( str, RTL_TEXTENCODING_UTF8 )
@@ -20,6 +55,8 @@ LayoutXMLFile::SearchL10NElements( XMLPa
         for ( ULONG i = 0; i < lst->Count(); i++ )
             if ( lst->GetObject( i )->GetNodeType() == XML_NODE_TYPE_ELEMENT )
                 HandleElement( ( XMLElement* )lst->GetObject( i ) );
+            else if ( lst->GetObject( i )->GetNodeType() == XML_NODE_TYPE_COMMENT )
+                lst->Remove( i-- );
 }
 
 std::vector<XMLAttribute*>
@@ -42,13 +79,14 @@ LayoutXMLFile::HandleElement( XMLElement
     
     if ( interesting.size() )
     {
-        ByteString id = BSTRING( interesting[0]->GetValue() );
+        std::vector<XMLAttribute*>::iterator i = interesting.begin();
+
+        ByteString id = BSTRING( (*i++)->GetValue() );
 
         if ( mMergeMode )
             InsertL10NElement( id, element );
         else
-            for ( std::vector<XMLAttribute*>::iterator i = ++interesting.begin();
-                  i != interesting.end(); ++i )
+            for ( ; i != interesting.end(); ++i )
             {
                 ByteString attributeId = id;
                 ByteString value = BSTRING( ( *i )->GetValue() );
Index: transex3/layout/layoutparse.hxx
===================================================================
RCS file: /cvs/l10n/transex3/layout/layoutparse.hxx,v
retrieving revision 1.2
retrieving revision 1.2.2.2
diff -p -u -u -p -b -w -B -r1.2 -r1.2.2.2
--- transex3/layout/layoutparse.hxx	6 Mar 2008 12:35:24 -0000	1.2
+++ transex3/layout/layoutparse.hxx	27 Mar 2008 13:09:53 -0000	1.2.2.2
@@ -1,3 +1,38 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
 #ifndef LAYOUTPARSE_HXX
 #define LAYOUTPARSE_HXX
 
@@ -14,6 +49,9 @@ public:
     BOOL Write( ByteString &aFilename );
     void HandleElement( XMLElement* element );
     void InsertL10NElement( ByteString const& id, XMLElement* element );
+
+    using XMLFile::InsertL10NElement;
+    using XMLFile::Write;
 };
 
 std::vector<XMLAttribute*> interestingAttributes( XMLAttributeList* lst );
Index: transex3/layout/makefile.mk
===================================================================
RCS file: /cvs/l10n/transex3/layout/makefile.mk,v
retrieving revision 1.2
retrieving revision 1.2.2.3
diff -p -u -u -p -b -w -B -r1.2 -r1.2.2.3
--- transex3/layout/makefile.mk	6 Mar 2008 12:36:23 -0000	1.2
+++ transex3/layout/makefile.mk	30 Mar 2008 13:36:03 -0000	1.2.2.3
@@ -1,3 +1,38 @@
+#*************************************************************************
+#
+#   OpenOffice.org - a multi-platform office productivity suite
+#
+#   $RCSfile$
+#
+#   $Revision$
+#
+#   last change: $Author$ $Date$
+#
+#   The Contents of this file are made available subject to
+#   the terms of GNU Lesser General Public License Version 2.1.
+#
+#
+#     GNU Lesser General Public License Version 2.1
+#     =============================================
+#     Copyright 2005 by Sun Microsystems, Inc.
+#     901 San Antonio Road, Palo Alto, CA 94303, USA
+#
+#     This library is free software; you can redistribute it and/or
+#     modify it under the terms of the GNU Lesser General Public
+#     License version 2.1, as published by the Free Software Foundation.
+#
+#     This library is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#     Lesser General Public License for more details.
+#
+#     You should have received a copy of the GNU Lesser General Public
+#     License along with this library; if not, write to the Free Software
+#     Foundation, Inc., 51 Franklin Street, 5th Floor, Boston,
+#     MA  02110-1301  USA
+#
+#*************************************************************************
+
 PRJ=..
 
 INCPRE=$(MISC)
@@ -20,14 +55,8 @@ CFLAGS+=-DSYSTEM_EXPAT
 
 # --- Files --------------------------------------------------------
 
-CXXFILES=\
-	layoutparse.cxx\
-	tralay.cxx
-
 APP1TARGET=$(TARGET)
 
-.IF "$(ENABLE_LAYOUT)" == "TRUE"
-
 OBJFILES =\
 	$(OBJ)/export2.obj\
 	$(OBJ)/helpmerge.obj\
@@ -41,27 +70,16 @@ APP1OBJS = $(OBJFILES)
 APP1STDLIBS =\
 	$(EXPATASCII3RDLIB)\
 	$(TOOLSLIB)\
-	$(VOSLIB)
+	$(VOSLIB)\
+	$(CPPULIB) \
+	$(SALLIB)	
 
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :  target.mk
 
 test .PHONY:
-	../$(INPATH)/bin/tralay -l en-US zoom.xml > out.sdf
+	../$(INPATH)/bin/tralay -l en-US -o out.sdf zoom.xml
 	cat out.sdf > trans.sdf
 	sed 's/en-US\t/de\tde:/' out.sdf >> trans.sdf 
-	../$(INPATH)/bin/tralay -m trans.sdf -l de zoom.xml #> zoom-DE.xml
-
-.ELSE # ENABLE_LAYOUT != TRUE
-
-$(BIN)$/$(APP1TARGET)$(EXECPOST):
-	touch $@
-
-ALLTAR:
-all:
-check:
-clean:
-
-.ENDIF # ENABLE_LAYOUT != TRUE
-
+	../$(INPATH)/bin/tralay -m trans.sdf -l de -o zoom-DE.xml zoom.xml
Index: transex3/layout/tralay.cxx
===================================================================
RCS file: /cvs/l10n/transex3/layout/tralay.cxx,v
retrieving revision 1.2
retrieving revision 1.2.2.7
diff -p -u -u -p -b -w -B -r1.2 -r1.2.2.7
--- transex3/layout/tralay.cxx	6 Mar 2008 12:36:58 -0000	1.2
+++ transex3/layout/tralay.cxx	4 Apr 2008 09:11:45 -0000	1.2.2.7
@@ -1,6 +1,43 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
 #include <com/sun/star/xml/sax/SAXException.hpp>
 #include <transex3/vosapp.hxx>
 
+#include <osl/file.hxx>
+
 #include "export.hxx"
 #include "layoutparse.hxx"
 #include "helpmerge.hxx"
@@ -12,7 +49,9 @@
 #define STRING( str ) String( str, RTL_TEXTENCODING_UTF8 )
 #define BSTRING( str ) ByteString( str, RTL_TEXTENCODING_UTF8 )
 
-using namespace ::rtl;
+using ::rtl::OUString;
+
+using namespace ::osl;
 using namespace ::com::sun::star;
 using namespace ::com::sun::star::uno;
 
@@ -31,7 +70,7 @@ class TranslateLayout : public Applicati
 
 public:
     TranslateLayout();
-    ~TranslateLayout();
+    virtual ~TranslateLayout();
     ByteString GetCommandLineParam( int i );
     ByteString GetOptionArgument( int const i );
     void ExceptionalMain();
@@ -40,6 +79,8 @@ public:
     void MergeLanguage( ByteString const& language );
     void ParseCommandLine();
     void CreateSDF();
+
+    using Application::GetCommandLineParam;
 };
 
 static void usage()
@@ -55,9 +96,29 @@ static void usage()
     exit( 2 );
 }
 
+static ByteString ConvertSystemPath( const ByteString& rPath )
+{
+    if( rPath.CompareTo( ".", 1 ) == 0 )
+    {
+        OUString sPath( rPath.GetBuffer(), rPath.Len(), RTL_TEXTENCODING_UTF8 );
+
+        ::rtl::OUString curDirPth, sURL;
+        osl_getProcessWorkingDir( &curDirPth.pData );
+
+        ::osl::FileBase::getAbsoluteFileURL( curDirPth, sPath, sURL );
+        ::osl::FileBase::getSystemPathFromFileURL( sURL, sPath );
+
+        return ByteString( rtl::OUStringToOString( sPath, RTL_TEXTENCODING_UTF8 ) );
+    }
+    else
+    {
+        return rPath;
+    }
+}
+
 ByteString TranslateLayout::GetCommandLineParam( int i )
 {
-    return ByteString( OUSTRING_CSTR( Application::GetCommandLineParam( i ) ) );
+    return ByteString( OUSTRING_CSTR( Application::GetCommandLineParam( sal::static_int_cast< USHORT >( i ) ) ) );
 }
 
 ByteString TranslateLayout::GetOptionArgument( int const i )
@@ -90,14 +151,14 @@ void TranslateLayout::ParseCommandLine()
             mLocalize = GetOptionArgument( ++i );
         }
         else if ( aParam.Equals( "-o" ) || aParam.Equals( "--output" ) )
-            mOutput = GetOptionArgument( ++i );
+            mOutput = ConvertSystemPath( GetOptionArgument( ++i ) );
         else if ( !aParam.CompareTo( "-", 1 ) )
         {
             fprintf( stderr, "error: No such option: %s\n", aParam.GetBuffer() );
             usage();
         }
         else
-            mFiles.push_back( aParam );
+            mFiles.push_back( ConvertSystemPath( aParam ) );
     }
     if ( !mFiles.size() )
     {
@@ -129,10 +190,14 @@ translateElement( XMLElement* element, B
                   ResData* resData, MergeDataFile& mergeData )
 {
     XMLAttributeList* attributes = element->GetAttributeList();
-    std::vector<XMLAttribute*> interesting = interestingAttributes( attributes );
-    ByteString id = BSTRING( interesting[0]->GetValue() );
-    for ( std::vector<XMLAttribute*>::iterator i = ++interesting.begin();
-          i != interesting.end(); ++i )
+    std::vector<XMLAttribute*> interesting( interestingAttributes( attributes ) );
+
+
+    if( !interesting.empty() )
+    {
+        std::vector<XMLAttribute*>::iterator i( interesting.begin() );
+        ByteString id = BSTRING( (*i++)->GetValue() );
+        for ( ; i != interesting.end(); ++i )
     {
         ByteString attributeId = id;
         attributeId += BSTRING ( **i );
@@ -154,24 +219,47 @@ translateElement( XMLElement* element, B
         }
     }
 }
+}
 
 static bool is_dir( ByteString const& name )
 {
-    if (DIR* dir = opendir( name.GetBuffer() ) )
+    DirectoryItem aItem;
+    OUString sFileURL( name.GetBuffer(), name.Len(), RTL_TEXTENCODING_UTF8 );
+    FileBase::getFileURLFromSystemPath( sFileURL, sFileURL );
+    if( DirectoryItem::get( sFileURL, aItem ) == FileBase::E_None )
+    {
+        FileStatus aStatus(FileStatusMask_Type);
+        if( aItem.getFileStatus( aStatus ) == FileBase::E_None )
     {
-        closedir( dir );
+            if( aStatus.getFileType() == FileStatus::Directory )
         return true;
     }
+    }
     return false;
 }
 
 static void make_directory( ByteString const& name )
 {
-#ifdef WNT
-    _mkdir( name.GetBuffer() );
-#else
-    mkdir( name.GetBuffer() , S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH );
-#endif
+    OUString sFileURL( name.GetBuffer(), name.Len(), RTL_TEXTENCODING_UTF8 );
+    FileBase::getFileURLFromSystemPath( sFileURL, sFileURL );
+    Directory::create( sFileURL );
+}
+
+static void insertMarker( XMLParentNode *p, ByteString const& file )
+{
+    if ( XMLChildNodeList* lst = p->GetChildList() )
+        if ( lst->Count() )
+        {
+            ULONG i = 1;
+            // Skip newline, if possible.
+            if ( lst->Count() > 1
+                 && lst->GetObject( 2 )->GetNodeType() == XML_NODE_TYPE_DEFAULT )
+                i++;
+            OUString marker = OUString::createFromAscii( "\n    NOTE: This file has been generated automagically by transex3/layout/tralay,\n          from source template: " )
+                + STRING( file )
+                + OUString::createFromAscii( ".\n          Do not edit, changes will be lost.\n" );
+            lst->Insert( new XMLComment( marker, 0 ), i );
+        }
 }
 
 void TranslateLayout::MergeLanguage( ByteString const& language )
@@ -190,6 +278,7 @@ void TranslateLayout::MergeLanguage( Byt
     }
     
     layoutXml->Extract();
+    insertMarker( layoutXml, xmlFile );
 
     ResData	resData( "", "", "" );
     resData.sResTyp = mProject; /* mGid1 ?? */
@@ -204,7 +293,11 @@ void TranslateLayout::MergeLanguage( Byt
                 translateElement( element, language, &resData, mergeData );
     }
     
+#ifndef WNT
     ByteString outFile = "/dev/stdout";
+#else
+    ByteString outFile = "\\\\.\\CON";
+#endif
     if ( mOutput.Len() )
     {
         outFile = mOutput;
@@ -235,7 +328,11 @@ void TranslateLayout::Merge()
 void TranslateLayout::CreateSDF()
 {
     ByteString xmlFile = mFiles.front();
+#ifndef WNT
     ByteString sdf = "/dev/stdout";
+#else
+    ByteString sdf = "\\\\.\\CON";
+#endif
     if ( mOutput.Len() )
         sdf = mOutput;
     Export::SetLanguages( mLanguages );
@@ -289,7 +386,7 @@ TranslateLayout::TranslateLayout()
     , mLocalize( "localize.sdf" )
     , mOutput()
     , mProject( "layout" )
-    , mRoot( "/root" )
+    , mRoot()
     , mMergeMode( false )
     , mLanguages()
     , mFiles()
Index: transex3/layout/zoom.xml
===================================================================
RCS file: /cvs/l10n/transex3/layout/zoom.xml,v
retrieving revision 1.2
retrieving revision 1.2.2.1
diff -p -u -u -p -b -w -B -r1.2 -r1.2.2.1
--- transex3/layout/zoom.xml	6 Mar 2008 12:37:10 -0000	1.2
+++ transex3/layout/zoom.xml	4 Apr 2008 09:11:45 -0000	1.2.2.1
@@ -1,4 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<!-- This is a template.  i18n translation is not performed in-place;
+     i18n translated xml files are generated from this template by
+     transex3/layout/tralay.  !-->
 
 <modaldialog xmlns="http://openoffice.org/2007/layout"
              xmlns:cnt="http://openoffice.org/2007/layout/container"
cvs diff: Diffing transex3/prj
cvs diff: Diffing transex3/scripts
cvs diff: Diffing transex3/source
cvs diff: Diffing transex3/workbench
